account_check_deposit/models/account_check_deposit.py

```diff
@@ -0,0 +1,368 @@
+            if line.payment_method_id.code == "manual" and line.payment_account_id:
+                account_id = line.payment_account_id.id
+                break
+        if not account_id:
+            account_id = self.company_id.account_journal_payment_debit_account_id.id
+        if not account_id:
+            raise UserError(
+                _("Missing 'Outstanding Receipts Account' on the company '%s'.")
+                % self.company_id.display_name
+            )
+        return {
+            "debit": total_debit,
+            "credit": 0.0,
+            "account_id": account_id,
+            "partner_id": False,
+            "currency_id": self.currency_id.id or False,
+            "amount_currency": total_amount_currency,
+        }
+
+    def validate_deposit(self):
+        am_obj = self.env["account.move"]
+        move_line_obj = self.env["account.move.line"]
+        for deposit in self:
+            move_vals = deposit._prepare_account_move_vals()
+            move = am_obj.create(move_vals)
+            total_debit = 0.0
+            total_amount_currency = 0.0
+            to_reconcile_lines = []
+            for line in deposit.check_payment_ids:
+                total_debit += line.debit
+                total_amount_currency += line.amount_currency
+                line_vals = self._prepare_move_line_vals(line)
+                line_vals["move_id"] = move.id
+                move_line = move_line_obj.with_context(
+                    check_move_validity=False
+                ).create(line_vals)
+                to_reconcile_lines.append(line + move_line)
+
+            # Create counter-part
+            counter_vals = deposit._prepare_counterpart_move_lines_vals(
+                total_debit, total_amount_currency
+            )
+            counter_vals["move_id"] = move.id
+            move_line_obj.create(counter_vals)
+            move.action_post()
+
+            deposit.write({"state": "done", "move_id": move.id})
+            for reconcile_lines in to_reconcile_lines:
+                reconcile_lines.reconcile()
+        return True
+
+    @api.onchange("company_id")
+    def onchange_company_id(self):
+        if self.company_id:
+            bank_journals = self.env["account.journal"].search(
+                [
+                    ("company_id", "=", self.company_id.id),
+                    ("type", "=", "bank"),
+                    ("bank_account_id", "!=", False),
+                ]
+            )
+            if len(bank_journals) == 1:
+                self.bank_journal_id = bank_journals[0]
+        else:
+            self.bank_journal_id = False
+
+    @api.onchange("journal_id")
+    def onchange_journal_id(self):
+        if self.journal_id:
+            if self.journal_id.currency_id:
+                self.currency_id = self.journal_id.currency_id
+            else:
+                self.currency_id = self.journal_id.company_id.currency_id
```

Comment:
I suppose in v15 we can go straight forwards but won't block for that 
```suggestion
        self.currency_id = self.journal_id.currency_id or self.journal_id.company_id.currency_id
```